app/src/main/java/eu/kanade/tachiyomi/data/track/anilist/AnilistInterceptor.kt

```
@@ -20,24 +20,20 @@ class AnilistInterceptor(private var refreshToken: String?) : Interceptor {
     override fun intercept(chain: Interceptor.Chain): Response {
         val originalRequest = chain.request()
 
-        if (refreshToken.isNullOrEmpty()) {
+        if (token.isNullOrEmpty()) {
             throw Exception("Not authenticated with Anilist")
         }
-
+        if (oauth == null){
+            oauth = anilist.loadOAuth()
+        }
         // Refresh access token if null or expired.
-        if (oauth == null || oauth!!.isExpired()) {
-            val response = chain.proceed(AnilistApi.refreshTokenRequest(refreshToken!!))
-            oauth = if (response.isSuccessful) {
-                Gson().fromJson(response.body()!!.string(), OAuth::class.java)
-            } else {
-                response.close()
-                null
-            }
+        if (oauth!!.isExpired()) {
+            anilist.logout()

```

I guess we should throw an exception here, since the requests are still going through.